web/src/main/java/org/openmrs/web/attribute/handler/ProgramFieldGenDatatypeHandler.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.Program;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.customdatatype.CustomDatatype.Summary;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.ProgramDatatype;
+import org.openmrs.serialization.SerializationException;
+import org.openmrs.serialization.SimpleXStreamSerializer;
+import org.springframework.stereotype.Component;
+
+/**
+ * Handler for the Program custom datatype
+ */
+@Component
```

Comment:
You don't need to "serialize" exactly. Just write out Program.uuid (and
read it back).

On Fri, Apr 24, 2015 at 11:45 AM, haolanchen <notifications@github.com>
wrote:

> In
> web/src/main/java/org/openmrs/web/attribute/handler/ProgramFieldGenDatatypeHandler.java
> <https://github.com/openmrs/openmrs-core/pull/1432#discussion_r29074754>:
>
> > +import javax.servlet.http.HttpServletRequest;
> > +
> > +import org.apache.commons.lang.StringUtils;
> > +import org.openmrs.Program;
> > +import org.openmrs.customdatatype.CustomDatatype;
> > +import org.openmrs.customdatatype.CustomDatatype.Summary;
> > +import org.openmrs.customdatatype.InvalidCustomValueException;
> > +import org.openmrs.customdatatype.datatype.ProgramDatatype;
> > +import org.openmrs.serialization.SerializationException;
> > +import org.openmrs.serialization.SimpleXStreamSerializer;
> > +import org.springframework.stereotype.Component;
> > +
> > +/**
> > + * Handler for the Program custom datatype
> > + */
> > +@Component
>
> Thanks man! I will take care of that. BTW, do you know which part of the
> Program data shall I serialize as xml?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1432/files#r29074754>.
>